### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by franzschug**
Land Cover Fraction Mapping Tutorial

### CODE REVIEW

File: docs/source/howto/lcf.rst [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+The Workflow
+-----------------------------------
+
+The workflow in this tutorial uses a series of submodules of the FORCE Higher Level Processing system (HLPS) to map sub-pixel fractions of land cover with Sentinel-2 imagery and synthetically mixed training data. 
+
+Some of these submodules have been described in other places of the FORCE documentation, and entire tutorials have been dedicated to others (respective links will be given where applicable).
+
+This tutorial illustrates the potential of FORCE to be used along the complete image processing chain, from downloading and pre-processing image acquisitions to producing meaningful spatial data. Its chapters correspond to the seven steps (and two optional steps) of the following workflow.
+
+This workflow is reproducible, as all commands, parameter files and intermediate data will be provided. Parameter files will be directly available for download throughout the workflow, while intermediate data will only be available in a data repository due to file size (Download data from Zenodo, doi: xxxxx)
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:27]: Probably add sth. like?

Variations of this workflow can also be used for regular classification or regression approaches, i.e. by skipping the synthetic mixing part.

File: docs/source/howto/lcf.rst [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+
+For processing, we use
+
+.. code-block:: bash
+
+   force-level2 /path/to/parametertile/directory/10_lcf_level_2.prm
+
+You can access the parameter file `here <../_static/parameter-files/tutorials/lcf/10_lcf_level_2.prm>`_ or use the one provided in the data repository. 	
+
+It is highly recommended to use a Digital Elevation Model (DEM) for topographic correction purposes. We here use a global SRTM/ASTER composite that we cannot provide for download. However, you can use any DEM of your choice here, for example the one provided by the `Copernicus Land Monitoring Service <https://www.eea.europa.eu/data-and-maps/data/copernicus-land-monitoring-service-eu-dem>`_. The use of a DEM is, however, not required to continue data processing, and as our example region is rather flat, the impact of missing topographic correction might be acceptable.
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:74]: mention global Copernicus DEM, and point to DEM tutorial? https://force-eo.readthedocs.io/en/latest/howto/dem.html

File: docs/source/howto/lcf.rst [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+
+
+Tip
+
+Please refer to the Level 2 ARD `tutorial <https://force-eo.readthedocs.io/en/latest/howto/l2-ard.html>`_ for further information about generating ARD in FORCE, and to the Digital Elevation Model `tutorial <https://force-eo.readthedocs.io/en/latest/howto/dem.html>_ for information on how to prepare a DEM for the FORCE processing system.
+
+Info
+
+--FORCE is also able to co-register Sentinel-2 data with Landsat time series data (`Tutorial <https://force-eo.readthedocs.io/en/latest/howto/coreg.html>`_. However, this workflow does not make use of this because it uniquely relies on Sentinel-2 data.
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:100]: remove leading -- and fix ()

File: docs/source/howto/lcf.rst [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+.. code-block:: bash
+
+	CSO = NUM MAX
+	
+You can access the parameter file `here <../_static/parameter-files/tutorials/lcf/11_lcf_cso.prm>`_ or use the one provided in the data repository.
+
+.. figure:: img/tutorial-lcf-cso.jpg
+   :height: 330
+
+   *Number of observations (A), average (B) and maximum (C) temporal distance in days between observations for our study period 01 Mar. 2018 and 30 Oct. 2018. Scale represents 96% of values.* |copy| *Franz Schug*
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:123]: AVG is not included in code block

File: docs/source/howto/lcf.rst [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+
+Based on CSO statistics, we assume that in most places of our study area, the number of observations (range from 10 to 42) and their temporal distance seems sufficient for generating robust spectral-temporal metrics.
+
+Data Aggregation / Spectral-Temporal Metrics
+-----------------------------------
+
+Previous research showed that sub-pixel fraction mapping of land cover with synthetically mixed training data can be performed using spectral-temporal metrics (STM) as input data.
+ 
+Spectral-temporal metrics are band-wise descriptive statistics, which summarize image reflectance (or an index derived thereof) of all valid observations within a defined time period, such as mean reflectance. They can be computed using the FORCE TSA submodule (tutorial).
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:132]: link is missing

File: docs/source/howto/lcf.rst [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+
+Compared to single observations, spectral-temporal metrics are able to increase model robustness across large areas as differences in acquisition dates across orbits has lower effects on the data. Spectral-temporal metrics are, in addition, able to represent phenological variation.
+
+In this tutorial, we use the first, second, and third quartile of reflectance of all clear-sky Sentinel-2 observations within our study period, as well as the 90th quantile and the standard deviation of two indicators of vegetation, Tasseled Cap Greenness and Normalized Difference Vegetation Index (NDVI).
+
+
+	*Spectral-temporal metrics of reflectance and vegetation indices (34 features) used in this example workflow.* |copy| *Franz Schug*
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:139]: is the indendation intended?

Comment (davidfrantz) [reply docs/source/howto/lcf.rst:139]: and is copyright necessary for this table?

File: docs/source/howto/lcf.rst [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+
+
+Sampling
+-----------------------------------
+
+Land cover fraction mapping with synthetically mixed training data requires spectral information from pure land cover surfaces to begin with, i.e., spectral reference data that can eventually be used to train a model that understands inter- and intra-class spectral variability. 
+
+Here, we are using reference information directly from the imagery (as opposed to, e.g., data from external spectral libraries or ground sampling). We identified 388 reference surfaces, i.e., pixels of 10x10 m that cover only a single land cover type, for five classes: Built-up surfaces (182 reference points), woody vegetation (70), non-woody vegetation (98), bare soil (15), and water (23).
+
+The number of reference points per class varies based on spectral intra-class variability (which is, e.g., higher for non-woody vegetation than for vegetation) and surface availability (e.g., few available reference points for bare soil).
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:198]: higher for non-woody vegetation than for vegetation.

is the 2x vegetation really what you mean?

File: docs/source/howto/lcf.rst [hunk 8]
```diff
@@ -0,0 +1,467 @@
...
+
+Regression-based spectral unmixing for land cover fraction mapping requires reference information about fractional land cover. This can be achieved, for example, by digitizing surface area types within a given pixel and use resulting fractional reference cover as input to regression model training. While this approach is very accurate, it is also time and labour intensive.
+
+We here use synthetically genererated fractional reference data for regression model training. These synthetically mixed data are based on known spectral information that represen pure surface types. For example, when we know what both a tree-covered surface and a road-covered surface spectrally look like in a Sentinel-2 image , we theoretically know what any kind of linear mixture between both surface look like, e.g., a pixel with 80% tree cover and 20% road cover, or a piuxel with 30% tree cover and 70% road cover.
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:250]: some minor spelling errors in this line

File: docs/source/howto/lcf.rst [hunk 9]
```diff
@@ -0,0 +1,467 @@
...
+
+ .. figure:: img/tutorial-lcf-mix-s2.png
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:252]: indent both the figure and the caption to make them appear in the rendered "quote block"?

File: docs/source/howto/lcf.rst [hunk 10]
```diff
@@ -0,0 +1,467 @@
...
+   :height: 360
+   
+*Spectral information of a pure tree-covered (green) and road-covered (red) pixel, as well as two different synthetic linear mixtures of both, for 10 `Sentinel-2 spectral bands <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-2-msi/resolutions/spatial>`_ as well as Tasseled Cap Greenness and NDVI. Dashed and dotted lines repreent linear mixtures.* |copy| *Franz Schug*
+
+This concept is, of course, transferable to our case, where we use spectral-temporal metrics instead of single observations. Please not that the number of features increases from 12 to 34, while the idea is the same. In this following case, the synthetic mixtures produce training data for 80% and 30% tree-covered surfaces, as well as, potentially 100% and 0% tree-covered surfaces.
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:257]: why potentially?

File: docs/source/howto/lcf.rst [hunk 11]
```diff
@@ -0,0 +1,467 @@
...
+
+Now use
+ 
+.. code-block:: bash
+	
+	force-magic-parameters -o /train parameterfiles/50_lcf_training.prm
+	
+to conveniently generate 15 parameter files (five per target class) representing all possible value combinations of the two replacement variables.
+
+We use a Support Vector Regression approach with a random 70/30 data split for training and internal model validation.
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:365]: can you shortly say that the logfiles contain all information for validation?

File: docs/source/howto/lcf.rst [hunk 12]
```diff
@@ -0,0 +1,467 @@
...
+
+Library completeness is not an established way to assess the quality of our training data library. However, it is a rough, but good indicator to show what surface types in the image might be under-represented in our training data, and, thus, in our reference points. In our example, we see that our training data represents forests really well (low values, dark areas). It seems like our training data does not as well represent some agricultural areas in the western part of our scene (higher values, brighter areas). Still, the maximum of our minimum MAE values in the image is ca. 150, which we consider low knowing that reflectance values can range from 0 to 10,000. Based on this, we do not see the necessity to identify further reference points.
+
+Info
+
+Note that low MAE values do not necessarily mean that the image pixels are correctly represented in the library. For example, in the case of spectral similarity of two different surface types, this algorithm cannot distinguish between correct and incorrect but similar spectral class representation.
+
+
+Model Training
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:336]: can you also link to the base parameterfile as you did for all other sections?

File: docs/source/howto/lcf.rst [hunk 13]
```diff
@@ -0,0 +1,467 @@
...
+
+.. |author-pic| image:: profile/fschug.jpg
+
++--------------+------------------------------------------------------------------------------------------+
++ |author-pic| + This tutorial was written by                                                             +
++              + Franz Schug,                                                                             +
++              + postdoc researcher at `SILVIS Lab <https://silvis.forest.wisc.edu/>`_.                   +
++              + *Views are his own.*                                                                     +
++--------------+------------------------------------------------------------------------------------------+
++ **EO**, **ARD**, **Data Science**, **Open Science**                                                     +
```
Comment (davidfrantz) [inline docs/source/howto/lcf.rst:466]: you can have your own tags if you like

**Comment by davidfrantz** [comment]
very nice tutorial, which covers the whole processing chain! This will be very handy for many people! Can you please have a look at the few comments, then we are ready for launch. Well done, thanks!

**Comment by franzschug** [comment]
Thanks for the feedback. I made according changes and implemented all your suggestions. The data now have a doi on Zenodo and are about to be uploaded (upload seems to take a while, but doi has been registered)

**Review by davidfrantz — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**